src/docker/api/models/response.cr

```diff
@@ -0,0 +1,20 @@
+require "json"
+
+module Docker::Api::Models
+  # :nodoc:
+  abstract struct Response
+    include JSON::Serializable
+
+    macro extended
+      macro finished
+        # Initializer based off {{@type.id}} properties
+        def initialize(
```

Comment:
This _may_ cause some odd behaviour if types extending this are then extended themselves. See https://forum.crystal-lang.org/t/constructor-variants-and-inheritance/3310/3.